crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,779 @@
+        out_dict.set_item("cache_max_entries", self.cache_max_entries)?;
+        out_dict.set_item("current_cache_entries", self.current_cache_entries)?;
+        out_dict.set_item("_cache_miss", self._cache_miss)?;
+        out_dict.set_item("_cache_hit", self._cache_hit)?;
+        out_dict.set_item("cache", self.cache.clone())?;
+        out_dict.set_item("library", self.library.library.to_object(py))?;
+        out_dict.set_item("gates", self.gates.clone())?;
+        Ok(out_dict.unbind())
+    }
+
+    fn __setstate__(&mut self, py: Python, state: PyObject) -> PyResult<()> {
+        let dict_state = state.downcast_bound::<PyDict>(py)?;
+        self.cache_max_entries = dict_state
+            .get_item("cache_max_entries")?
+            .unwrap()
+            .extract()?;
+        self.current_cache_entries = dict_state
+            .get_item("current_cache_entries")?
+            .unwrap()
+            .extract()?;
+        self._cache_miss = dict_state.get_item("_cache_miss")?.unwrap().extract()?;
+        self._cache_hit = dict_state.get_item("_cache_hit")?.unwrap().extract()?;
+        self.library = CommutationLibrary {
+            library: dict_state.get_item("library")?.unwrap().extract()?,
+        };
+        self.cache = dict_state.get_item("cache")?.unwrap().extract()?;
+        self.gates = dict_state.get_item("gates")?.unwrap().extract()?;
+        Ok(())
+    }
+}
+
+impl CommutationChecker {
+    #[allow(clippy::too_many_arguments)]
+    fn commute_inner(
+        &mut self,
+        py: Python,
+        op1: &OperationRef,
+        params1: &[Param],
+        attrs1: Option<&ExtraInstructionAttributes>,
+        qargs1: &[Qubit],
+        cargs1: &[Clbit],
+        op2: &OperationRef,
+        params2: &[Param],
+        attrs2: Option<&ExtraInstructionAttributes>,
+        qargs2: &[Qubit],
+        cargs2: &[Clbit],
+        max_num_qubits: u32,
+    ) -> PyResult<bool> {
+        if let Some(gates) = &self.gates {
+            if !gates.is_empty() && (!gates.contains(op1.name()) || !gates.contains(op2.name())) {
+                return Ok(false);
+            }
+        }
+
+        let commutation: Option<bool> = commutation_precheck(
+            op1,
+            params1,
+            attrs1,
+            qargs1,
+            cargs1,
+            op2,
+            params2,
+            attrs2,
+            qargs2,
+            cargs2,
+            max_num_qubits,
+        );
+        if let Some(is_commuting) = commutation {
+            return Ok(is_commuting);
+        }
+
+        let reversed = if op1.num_qubits() != op2.num_qubits() {
+            op1.num_qubits() > op2.num_qubits()
+        } else {
+            // we use a conversion to BigInt here instead of just op1.name() >= op2.name(), because
+            // strings are sorted differently in Rust and in Python,
+            BigInt::from_signed_bytes_be(op1.name().as_bytes())
+                >= BigInt::from_signed_bytes_be(op2.name().as_bytes())
```

Comment:
Instead of building a BigInt, just compare both the `&str` and the length of the name.
```suggestion
            (op1.name().len(), op1.name()) >= (op2.name().len(), op2.name())
```
Seems to have the same effect.